plopfile.js

```diff
@@ -0,0 +1,320 @@
+  const componentExists = comp => {
+    const pageComponents = fs.readdirSync('app/components');
+    const pageContainers = fs.readdirSync('app/containers');
+    const components = pageComponents.concat(pageContainers);
+
+    return components.indexOf(comp) >= 0;
+  };
+
+  // Stateless component generator
+  plop.setGenerator('component', {
+    description: 'Generate a Component either using ES6 class or Stateless Function',
+    prompts: [{
+      type: 'list',
+      name: 'type',
+      message: 'Select the type of component',
+      choices: () => ['ES6 Class', 'Stateless Function']
+    }, {
+      type: 'input',
+      name: 'name',
+      message: 'What should it be called?',
+      validate: value => {
+        if ((/.+/).test(value)) {
+          return true;
+        }
+        return 'name is required';
+      }
+    }, {
+      type: 'input',
+      name: 'description',
+      message: 'Describe what the component does?',
+      validate: value => {
+        if ((/.+/).test(value)) {
+          return true;
+        }
+        return 'description is required';
+      }
+    }, {
+      type: 'confirm',
+      name: 'wantCSS',
+      message: 'Do you want to create corresponding CSS file?'
+    }],
+    actions: data => {
+      const actions = [{
+        type: 'add',
+        path: 'app/components/{{properCase name}}/index.js',
+        templateFile: data.type === 'ES6 Class' ? './webpack/templates/component/es6.js.hbs' : './webpack/templates/component/stateless.js.hbs',
+        abortOnFail: true
+      }, {
+        type: 'add',
+        path: 'app/components/{{properCase name}}/{{properCase name}}.test.js',
+        templateFile: './webpack/templates/component/component.test.js.hbs',
+        abortOnFail: true
+      }];
+
+      if (data.wantCSS) {
+        actions.push({
+          type: 'add',
+          path: 'app/components/{{properCase name}}/styles.css',
+          templateFile: './webpack/templates/component/styles.css.hbs',
+          abortOnFail: true
+        });
+      }
+
+      return actions;
+    }
+  });
+
+  // Container Component Generator
+  plop.setGenerator('container', {
+    description: 'Generate a Container Component',
+    prompts: [{
+      type: 'input',
+      name: 'name',
+      message: 'What should it be called?',
+      validate: value => {
+        if ((/.+/).test(value)) {
+          return true;
+        }
+        return 'name is required';
+      }
+    }, {
+      type: 'input',
+      name: 'description',
+      message: 'Describe what the container component does?',
+      validate: value => {
+        if ((/.+/).test(value)) {
+          return true;
+        }
+        return 'description is required';
+      }
+    }, {
+      type: 'confirm',
+      name: 'wantCSS',
+      message: 'Do you want to create corresponding CSS file?'
+    }, {
+      type: 'confirm',
+      name: 'connectRedux',
+      message: 'Do you want to connect the page to a redux store/dispatch actions?'
```

Comment:
We don't need to ask this here, because if they chose to generate a `container` they WANT to be connected to the store – otherwise it's not a `container` but a `component`!